### PR TITLE
cherry-pick for 0.13: fix(framework): omit detail property of errors

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:
This cherry-picks https://github.com/garden-io/garden/commit/24f32fad to `0.13`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/events.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ export interface CommandInfoPayload extends CommandInfo {
 export function toGraphResultEventPayload(result: GraphResult): GraphResultEventPayload {
   const payload = sanitizeValue({
     ...omit(result, "dependencyResults", "task"),
-    error: result.error ? String(result.error) : null,
+    error: result.error ? String(omit(result.error, "detail")) : null,
```
Comment (stefreak) [inline core/src/events.ts:102]: I think with `String()` the omit of detail is not necessary anymore.

```
String(new Error("foo"))
'Error: foo'
```

**Review by stefreak — CHANGES_REQUESTED** [review]

**Comment by stefreak** [comment]
@vvagaytsev I think we can close this PR

File: core/src/events.ts (hunk 1 above)
Comment (stefreak) [reply core/src/events.ts:102]: In my experiments, for an object like `String({})` yields `[Object object]` and `String(new Error("message"))` yields `Error: message`

Comment (stefreak) [reply core/src/events.ts:102]: In the original commit, it was assigned directly without `String`: https://github.com/garden-io/garden/pull/3623/files – so the change became obsolete

Comment (vvagaytsev) [reply core/src/events.ts:102]: Ok, thanks!

**Pull request closed without merging**